src/c2goto/library/pthread_lib.c

```diff
@@ -483,23 +582,90 @@ __ESBMC_HIDE:;
   return 0;
 }
 
+// The pthread_key_create() function shall create a thread-specific
+// data key visible to all threads in the process.
+// source: https://linux.die.net/man/3/pthread_key_create
 int pthread_key_create(pthread_key_t *key, void (*destructor)(void *))
 {
 __ESBMC_HIDE:;
-  // TODO
-  return 0;
+  __ESBMC_atomic_begin();
+  int result = 0;
+  __ESBMC_assert(
+    key != NULL,
+    "In pthread_key_create, key parameter must be different than NULL.");
+  // the value NULL shall be associated with the new key in all active threads
+  result = insert_key_value(mylist, __ESBMC_next_thread_key, NULL);
+  __ESBMC_thread_key_destructors[__ESBMC_next_thread_key] = destructor;
+  // store the newly created key value at *key
+  *key = __ESBMC_next_thread_key++;
+  if(result < 0)
```

Comment:
Add a comment here explaining when the result would be < 0 in this case.

Reply:
Fixed.